tracer/src/Datadog.Trace/Activity/Handlers/DefaultActivityHandler.cs

```diff
@@ -43,22 +43,25 @@ public void ActivityStarted<T>(string sourceName, T activity)
             ulong? spanId = null;
             string? rawTraceId = null;
             string? rawSpanId = null;
-
             if (activity is IW3CActivity w3cActivity)
             {
                 // If the user has specified a parent context, get the parent Datadog SpanContext
                 if (w3cActivity.ParentSpanId is not null
-                    && w3cActivity.ParentId is string parentId
-                    && ActivityMappingById.TryGetValue(parentId, out ActivityMapping mapping))
+                 && w3cActivity.ParentId is { } parentId)
                 {
-                    parent = mapping.Scope.Span.Context;
+                    if (ActivityMappingById.TryGetValue(parentId, out ActivityMapping mapping))
+                    {
+                        parent = mapping.Scope.Span.Context;
+                    }
+                    else
+                    {
+                        // create a new parent span context for the ActivityContext
+                        parent = Tracer.Instance.CreateSpanContext(SpanContext.None, rawTraceId: activity.ParentId, rawSpanId: w3cActivity.ParentSpanId);
```

Comment:
To fix the traceId issue you're having, I think you'll need to create a new SpanContext as pass it in as the first argument. The spanId you should be able to just convert from hex string to ulong, but the traceId you'll have to take the last 16 hex chars and just convert that hex string to ulong